### PR TITLE
emit exceptions too, some dedupe, less strict diff

### PR DESCRIPTION
**Pull request by olydis**

### CODE REVIEW

**Comment by olydis** [comment]
# publish job
~~~ Haskell
> waiting for PR to get merged or closed
~~~

**Comment by olydis** [comment]
> win32-x64 restart

**Comment by olydis** [comment]
@fearthecowboy read lock stuff http://autorestci1.blob.core.windows.net/autorest-ci/CIe9n76_autorest_2651_20171011T004842

**Comment by olydis** [comment]
> win32-x64 restart

**Comment by olydis** [comment]
> win32-x64 restart

**Comment by olydis** [comment]
cool thanks! Got it again but slightly different after nuking `.autorest` and the lock file next to it: http://autorestci1.blob.core.windows.net/autorest-ci/CIe9n76_autorest_2651_20171011T013822

File: Samples/1a-code-generation-minimal/Client/Models/Category.cs [hunk 1]
```diff
@@ -1,90 +1 @@
...
-            string resultName;
-            if (deserializeName(payload, "name", out resultName))
-            {
-                result.Name = resultName;
-            }
-            return result;
-        }
-    }
-}
+SRC
```
Comment (ahmedalejo) [inline Samples/1a-code-generation-minimal/Client/Models/Category.cs]: what´s SRC? i think commit messed the source files up

**Pull request merged**